nexus/src/app/image.rs

```diff
@@ -24,54 +25,77 @@ use omicron_common::api::external::LookupResult;
 use omicron_common::api::external::LookupType;
 use omicron_common::api::external::NameOrId;
 use omicron_common::api::external::ResourceType;
+use omicron_common::api::external::UpdateResult;
 use std::str::FromStr;
 use std::sync::Arc;
 use uuid::Uuid;
 
 impl super::Nexus {
-    pub fn image_lookup<'a>(
+    pub async fn image_lookup<'a>(
         &'a self,
         opctx: &'a OpContext,
         image_selector: params::ImageSelector,
-    ) -> LookupResult<lookup::Image<'a>> {
+    ) -> LookupResult<ImageLookup<'a>> {
         match image_selector {
             params::ImageSelector {
                 image: NameOrId::Id(id),
                 project: None,
             } => {
-                let image = LookupPath::new(opctx, &self.db_datastore)
-                    .image_id(id);
-                Ok(image)
+                let (.., db_image) = LookupPath::new(opctx, &self.db_datastore)
+                    .image_id(id).fetch().await?;
+                let lookup = match db_image.project_id {
+                    Some(_) => ImageLookup::ProjectImage(LookupPath::new(opctx, &self.db_datastore)
+                        .project_image_id(id)),
+                    None => {
+                        ImageLookup::SiloImage(LookupPath::new(opctx, &self.db_datastore)
+                            .silo_image_id(db_image.silo_id))
+                    },
+                };
```

Comment:
This is a bit unique. Because there's only a single endpoint for lookup images by id (`/v1/images/{image_id}`) we don't actually know if it's a silo or project image when we get the ID. We first have to lookup the image and see if `project_id` is populated. If it is, we know it's a project id. Otherwise it's a silo id.